### DISCUSSION TITLE
Discussion: Can vcpkg use the compiler specified from the CMake command line arguments in manifest mode?

### DISCUSSION THREAD

**Discussion by bowenxuuu** [microsoft/vcpkg#41140] [General]
In my screenshot, I have selected the `Clang 19.1.0 x86_64-pc-windows-msvc` kit, but `vcpkg` always uses `MSVC (cl.exe)` to compile the third-party libraries. However, my own source code will be compiled with `clang++.exe`.

Although the entire compilation process completes without errors, and the program can run, I am still concerned about potential ABI compatibility issues between the third-party libraries and my own source code, because they are compiled with two different compilers.

BTW, I use `LLVM (Clang)` because tools like `clangd`, `clang-tidy`, `clang-format`, etc., are helpful for me.

![aaa](https://github.com/user-attachments/assets/6f902909-fca5-42bf-8a5b-1d451a038f56)

**Comment by autoantwort** [comment]
No. You have to write a toolchain file and a custom triplet. See https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_chainload_toolchain_file

**Reply by Neumann-A** [reply]
> No. You have to write a toolchain file and a custom triplet

or you generate a custom triplet with the settings of the calling cmake process. Requires writing a custom vcpkg toolchain however

**Reply by bowenxuuu** [reply]
It's a bit difficult for me. I think it requires a very good understanding and familiarity with CMake.
